### DISCUSSION TITLE
What technology does Cordova use to enable JavaScript to call Android or iOS native methods？Is that JSBridge

### DISCUSSION THREAD

**Discussion by EmptyPages** [apache/cordova#445] [General]
What technology does Cordova use to enable JavaScript to call Android or iOS native methods？Is that JSBridge

**Comment by breautek** [comment]
There isn't anything really called the "JSBridge"

On iOS it's mostly [WKUserContentController](https://developer.apple.com/documentation/webkit/wkusercontentcontroller?language=objc) which handles webview -> native messages, then I believe native -> webview messages are handled via evaluating JS.

iOS 14+ has a promise-based API where content controllers can reply back with a message asynchronously, but Cordova supports iOS 11+ so it doesn't use that API.

On Android, it uses javascript interfaces, it allows you to bind a java method as a js function which is exposed in the JS environment, it supports some JS basic types and allows you to synchronously return basic JS types as well. See [JavascriptInterface](https://developer.android.com/reference/android/webkit/JavascriptInterface).
For continuous callbacks I believe cordova-android evaluates js onto a pre-existing global function setup by the cordova, like iOS does.
